docs/api/web-contents.md

```diff
@@ -934,6 +934,21 @@ Returns `string` - The title of the current web page.
 
 Returns `boolean` - Whether the web page is destroyed.
 
+#### `contents.close([opts])`
+
+* `opts` Object (optional)
+  * `waitForBeforeUnload` boolean - if true, fire the `beforeunload` event
+    before closing the page. If the page prevents the unload, the WebContents
+    will not be closed. The [`will-prevent-unload`](#event-will-prevent-unload)
+    will be fired if the page requests prevention of unload.
```

Comment:
Is this option needed? What problem is solved by adding this option instead of always firing `beforeunload` and honoring event.preventDefault()?

It feels a little like we're introducing duelling APIs: e.g. "closes, except prevent can override closing, except contents.close() can override that override"

Reply:
`waitForBeforeUnload: true` will do a round trip to the renderer and invoke `beforeunload`, whereas not doing that will instantly close the webContents without a roundtrip and without invoking the method.